packages/dds/map/src/mapKernel.ts

```diff
@@ -693,14 +1053,20 @@ export class MapKernel {
 					return;
 				}
 				this.deleteCore(op.key, local);
+				// Adjust the keys order if the deleted key is already ack'd
+				this.ackedKeysIndexTracker.delete(op.key);
 			},
 			submit: (op: IMapDeleteOperation, localOpMetadata: MapKeyLocalOpMetadata) => {
 				this.resubmitMapKeyMessage(op, localOpMetadata);
 			},
 			applyStashedOp: (op: IMapDeleteOperation) => {
 				// We don't reuse the metadata pendingMessageId but send a new one on each submit.
 				const previousValue = this.deleteCore(op.key, true);
-				return createKeyLocalOpMetadata(op, this.getMapKeyMessageId(op), previousValue);
+				const previousIndex = this.deleteKeysIndex(op.key);
+				this.incrementPendingDeleteCount(op.key);
+				const messageId = this.getMapKeyMessageId(op);
+				this.updatePendingSetIds(op, messageId);
+				return createKeyLocalOpMetadata(op, messageId, previousValue, previousIndex);
 			},
 		});
```

Comment:
How is the test coverage for this w.r.t applyStashedOp? I'd imagine the existing tests are somewhat sparse and probably don't test iteration order much.

Reply:
I added one unit test starting from line 423 of the file `map.order.spec.ts`, but yeah probably we need more test suites about this